cascade-prelude/src/Cascade/Log/Message.hs

```diff
@@ -0,0 +1,121 @@
+{-|
+Module      : Cascade.Log.Message
+Description : !!! INSERT MODULE SHORT DESCRIPTION !!!
+Copyright   : (c) 2020-2021 Cascade
+License     : MPL 2.0
+Maintainer  : Mohammad Hasani <the-dr-lazy@pm.me> (the-dr-lazy.github.io)
+Stability   : Stable
+Portability : POSIX
+
+!!! INSERT MODULE LONG DESCRIPTION !!!
+-}
+
+module Cascade.Log.Message (log, Message, Scope(..), fmtMessage, logMessageStdoutAndStderr, logMessageStderr, logMessageStdout) where
+
+import           Cascade.Log.Severity                ( Severity(..) )
+import           Chronos                             ( Time )
+import qualified Chronos
+import qualified Chronos.Locale.English             as Chronos
+import           Colog.Actions                       ( logTextStderr
+                                                     , logTextStdout
+                                                     )
+import           Colog.Core                          ( LogAction )
+import           Colog.Core.Action                   ( cfilter )
+import           Colog.Monad                         ( WithLog
+                                                     , logMsg
+                                                     )
+import qualified Data.Text                          as Text
+import qualified Data.Text.Lazy.Builder             as TB
+import qualified Data.Text.Lazy.Builder.Int         as TB
+import qualified Data.Vector                        as Vector
+import           GHC.Stack                           ( SrcLoc(..) )
+import           System.Console.ANSI                 ( Color(..)
+                                                     , ColorIntensity(Vivid)
+                                                     , ConsoleLayer(Foreground)
+                                                     , SGR(..)
+                                                     , setSGRCode
+                                                     )
+
+data Scope = Cli | Api
+  deriving stock Eq
+
+data Message = Message
+  { message  :: Text
+  , severity :: Severity
+  , scope    :: Scope
+  , time     :: Time
+  , location :: CallStack
+  }
+
+log :: WithLog env Message m => MonadIO m => Scope -> Severity -> Text -> m ()
+log scope severity message = do
+  time <- liftIO Chronos.now
+  withFrozenCallStack (logMsg Message { location = callStack, .. })
+
+fmtMessage :: Message -> Text
```

Comment:
In other places, we are using the `prettyPrint` instead of `fmt`. Let's use it as a convention.